drivers/hmis/app/controllers/ac_hmis/reports_controller.rb

```diff
@@ -0,0 +1,25 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module AcHmis
+  class ReportsController < Hmis::BaseController
+    def prevention_assessment_report
+      report = AcHmis::ReportApi.new.prevention_assessment_report(referral_id: params[:referral_id])
+
+      render body: report.body, status: report.http_status, content_type: 'application/pdf'
+    end
+
+    def consumer_summary_report
+      client = Hmis::Hud::Client.find_by(id: params[:client_id])
+      umci = client&.ac_hmis_mci_unique_id&.value
+      return render body: "No UMCI found for client with ID '#{params[:client_id]}'", status: 404 unless umci.present?
```

Comment:
It can happen that a client doesn't have a UMCI yet because they were recently created. Let's update this message:
```suggestion
      return render body: "An MCI Unique ID has not yet been assigned to Client #{params[:client_id]}. Please check back in 24 hours.", status: 404 unless umci.present?
```